components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java

```diff
@@ -834,7 +835,35 @@ protected void unsetOAuth2RevocationProcessor(OAuth2RevocationProcessor oAuth2Re
         if (log.isDebugEnabled()) {
             log.debug("Unset Oauth2 revocation processor.");
         }
-        OAuth2ServiceComponentHolder.setOrganizationUserResidentResolverService(null);
+        OAuth2ServiceComponentHolder.getInstance().setRevocationProcessor(null);
+    }
+
+    @Reference(
+            name = "oauth2.token.persistence.handler",
+            service = IndirectTokenRevocationHandler.class,
+            cardinality = ReferenceCardinality.OPTIONAL,
```

Comment:
Since cardinality is optional anyone looking at this would think that the system can run without a IndirectTokenRevocationHandler. Since that is not the case shall we add a comment inside the method mentioning that if now handler is engaged a default will be set.

Reply:
This is since the default implementation is used if and only if there is no extended impl registered. Need the cardinality as `ReferenceCardinality.OPTIONAL` to avoid having errors at startup. Other than that, there will be an implementation always. 
(Removed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/commit/69f8c55fb76f15cbfe22b3c551b17491dc1796b0 )